### PR TITLE
feat: product extension form

### PR DESCRIPTION
**Pull request by foolmonkey**
Adds basic UI for the Product form in the extension popup.
- No full validation, just the UI and form structure using React Hook Form.
- TODO comments left in to add extra functionality

image

### CODE REVIEW

File: apps/extension/package.json [hunk 1]
```diff
@@ -6,13 +6,16 @@
   "scripts": {
     "dev": "vite",
     "build": "tsc -b && vite build",
+    "watch": "concurrently \"tsc -b -w\" \"vite build --watch\"",
```
Comment (foolmonkey) [inline apps/extension/package.json:9]: This is just for convenience really, so I could have a watch script running in the BG while I code the extension and have it show up real-time. Installed a package Concurrently globally to make use of it.

File: apps/extension/public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        "system.storage",
+        "tabCapture",
+        "tabGroups",
+        "tabs",
+        "topSites",
+        "webNavigation",
+        "webRequest",
+        "webRequestBlocking"
+      ]
+}
```
Comment (foolmonkey) [inline apps/extension/public/manifest.json:46]: We'll remove these as permissions we find out what we actually do need. Leaving them for convenience for now.

File: apps/extension/src/pages/ProductForm/ProductForm.tsx
Comment (foolmonkey) [inline apps/extension/src/pages/ProductForm/ProductForm.tsx:1]: Probably can refactor certain Form field into their own component as complexity grows

File: apps/extension/package.json (hunk 1 above)
Comment (iamnotrodger) [reply apps/extension/package.json:9]: hehe i yes i've used in the past

File: apps/extension/public/manifest.json (hunk 2 above)
Comment (iamnotrodger) [reply apps/extension/public/manifest.json:46]: yeah lol no idea what we need

**Review by iamnotrodger — APPROVED** [review]
looks sliiiccckkk

**Review by iamnotrodger — COMMENTED** [review]
oh lol we both used zod

**Comment by iamnotrodger** [comment]
hmmm lets keep it simple for now but lets plan on converging these types somewhere

**Review by iamnotrodger — COMMENTED** [review]
hmmm actual perchance merge my pr first and create or a derivation of the product schema here

**Pull request merged**